internal/node/node.go

```diff
@@ -83,7 +76,6 @@ func (dn *Node) SpawnDescendant() *Node {
 	return &Node{
 		Parent: dn,
 		OTEL:   dn.OTEL,
-		Span:   dn.Span,
```

Comment:
I like the idea of relying on an existing library to track spans, but at the same time, this PR sort of seems like it makes the logic more complex to follow. It now feels like there's almost a circular dependency between nodes and contexts whereas before nodes could operate independent of contexts

Reply:
not sure if you're picking the term `circular` loosely, or you're actually seeing circular behavior.  My hope is that the `node` package is an optional passthrough when handling otel stuff.  And on that note, all the otel behavior inside `/internal/node` should probably get moved to something like `/internal/otelutils`.  It's in a weird crossroads in general, since it has grown organically out of the node design.

Reply:
Lemme take that on as a follow up.  I don't think it should be blocking here, but it _is_ awkward, and I'd like to get it cleaned up sooner rather than later.